walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,71 @@
 
 package com.google.sps;
 
+import java.util.ArrayList; 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;   
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      
+    ArrayList<TimeRange> event_time = new ArrayList<TimeRange>();
+    Set<String> request_attendees = new HashSet<String>(request.getAttendees());
+    for (Event event: events) {
+        Set<String> intersection_attendees = new HashSet<String>(request.getAttendees());
+        Set<String> event_attendees = new HashSet<String>(event.getAttendees());
+        intersection_attendees.retainAll(event_attendees);
+        if (!intersection_attendees.isEmpty()) {
+            event_time.add(event.getWhen());
+        }
+    }
+    return queryByTime(mergeEventTime(event_time), request.getDuration());
   }
+
+  private ArrayList<TimeRange> mergeEventTime(ArrayList<TimeRange> event_time) {
+
+      int size = event_time.size();
+      if (size < 2) return event_time;
+      
+      Collections.sort(event_time, TimeRange.ORDER_BY_START);
+      ArrayList<TimeRange> event_time_start_order = new ArrayList<TimeRange>(event_time);
+      Collections.sort(event_time, TimeRange.ORDER_BY_END);
```

Comment:
is Collections.sort stable?

Reply:
I think it is stable as Collections.sort uses mergeSort and will not change the original relative position of two equal element. But whether or not this sort is stable would not affect the result I think.

Reply:
I see. I misunderstood your algorithm. Reading it again, you don't need the 2 sort to be stable, as you are only using one endpoint of the range in each list. 

Looks good me.